From 0a5481977f6e8a8f50684abfabdc12fc535d1c58 Mon Sep 17 00:00:00 2001
From: Matthew Collins <me@thinkof.name>
Date: Tue, 19 Mar 2019 23:07:07 +0000
Subject: [PATCH] clang-cl build fixes

---
 src/lj_def.h | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

diff --git a/src/lj_def.h b/src/lj_def.h
index e67bb24..e44999e 100644
--- a/src/lj_def.h
+++ b/src/lj_def.h
@@ -262,19 +262,16 @@ static LJ_AINLINE uint32_t lj_fls(uint32_t x)
   return _CountLeadingZeros(x) ^ 31;
 }
 #else
-unsigned char _BitScanForward(uint32_t *, unsigned long);
-unsigned char _BitScanReverse(uint32_t *, unsigned long);
-#pragma intrinsic(_BitScanForward)
-#pragma intrinsic(_BitScanReverse)
+#include <intrin.h>
 
 static LJ_AINLINE uint32_t lj_ffs(uint32_t x)
 {
-  uint32_t r; _BitScanForward(&r, x); return r;
+  unsigned long r; _BitScanForward(&r, x); return r;
 }
 
 static LJ_AINLINE uint32_t lj_fls(uint32_t x)
 {
-  uint32_t r; _BitScanReverse(&r, x); return r;
+  unsigned long r; _BitScanReverse(&r, x); return r;
 }
 #endif
 
-- 
2.21.0

